### ISSUE TITLE
WebGL freezes in `gl.drawArraysInstanced` in OSMesa

### ISSUE THREAD

**Issue by asajeffrey** [servo/servo#26001]
Running `RUST_LOG=canvas LIBGL_ALWAYS_SOFTWARE=1 ./mach test-wpt /_webgl/conformance2/transform_feedback/too-small-buffers.html` on Linux produces:
```
...
0:05.51 pid:21232 <h1>integer overflow</h1>
 0:05.51 pid:21232 [2020-03-20T16:09:10Z TRACE canvas::webgl_thread] processing WebGLCommand(WebGLContextId(1), UseProgram(Some(WebGLProgramId(1))), WebGLCommandBacktrace)
 0:05.51 pid:21232 [2020-03-20T16:09:10Z DEBUG canvas::webgl_thread] WebGLImpl::apply(UseProgram(Some(WebGLProgramId(1))))
 0:05.51 pid:21232 [2020-03-20T16:09:10Z TRACE canvas::webgl_thread] processing WebGLCommand(WebGLContextId(1), BindBuffer(34962, Some(WebGLBufferId(1))), WebGLCommandBacktrace)
 0:05.51 pid:21232 [2020-03-20T16:09:10Z DEBUG canvas::webgl_thread] WebGLImpl::apply(BindBuffer(34962, Some(WebGLBufferId(1))))
 0:05.51 pid:21232 [2020-03-20T16:09:10Z TRACE canvas::webgl_thread] processing WebGLCommand(WebGLContextId(1), BufferData(34962, IpcBytesReceiver { os_receiver: OsIpcReceiver { fd: Cell { value: 126 } } }, 35041), WebGLCommandBacktrace)
 0:05.51 pid:21232 [2020-03-20T16:09:10Z DEBUG canvas::webgl_thread] WebGLImpl::apply(BufferData(34962, IpcBytesReceiver { os_receiver: OsIpcReceiver { fd: Cell { value: 126 } } }, 35041))
 0:05.51 pid:21232 [2020-03-20T16:09:10Z TRACE canvas::webgl_thread] processing WebGLCommand(WebGLContextId(1), BindBuffer(34962, Some(WebGLBufferId(2))), WebGLCommandBacktrace)
 0:05.51 pid:21232 [2020-03-20T16:09:10Z DEBUG canvas::webgl_thread] WebGLImpl::apply(BindBuffer(34962, Some(WebGLBufferId(2))))
 0:05.51 pid:21232 [2020-03-20T16:09:10Z TRACE canvas::webgl_thread] processing WebGLCommand(WebGLContextId(1), BufferData(34962, IpcBytesReceiver { os_receiver: OsIpcReceiver { fd: Cell { value: 130 } } }, 35041), WebGLCommandBacktrace)
 0:05.51 pid:21232 [2020-03-20T16:09:10Z DEBUG canvas::webgl_thread] WebGLImpl::apply(BufferData(34962, IpcBytesReceiver { os_receiver: OsIpcReceiver { fd: Cell { value: 130 } } }, 35041))
 0:05.51 pid:21232 [2020-03-20T16:09:10Z TRACE canvas::webgl_thread] processing WebGLCommand(WebGLContextId(1), BindBufferBase(35982, 0, Some(WebGLBufferId(3))), WebGLCommandBacktrace)
 0:05.51 pid:21232 [2020-03-20T16:09:10Z DEBUG canvas::webgl_thread] WebGLImpl::apply(BindBufferBase(35982, 0, Some(WebGLBufferId(3))))
 0:05.51 pid:21232 [2020-03-20T16:09:10Z TRACE canvas::webgl_thread] processing WebGLCommand(WebGLContextId(1), BufferData(35982, IpcBytesReceiver { os_receiver: OsIpcReceiver { fd: Cell { value: 131 } } }, 35041), WebGLCommandBacktrace)
 0:05.51 pid:21232 [2020-03-20T16:09:10Z DEBUG canvas::webgl_thread] WebGLImpl::apply(BufferData(35982, IpcBytesReceiver { os_receiver: OsIpcReceiver { fd: Cell { value: 131 } } }, 35041))
 0:05.51 pid:21232 [2020-03-20T16:09:10Z TRACE canvas::webgl_thread] processing WebGLCommand(WebGLContextId(1), DrawingBufferWidth(WebGLSender(..)), WebGLCommandBacktrace)
 0:05.51 pid:21232 [2020-03-20T16:09:10Z DEBUG canvas::webgl_thread] WebGLImpl::apply(DrawingBufferWidth(WebGLSender(..)))
 0:05.51 pid:21232 [2020-03-20T16:09:10Z TRACE canvas::webgl_thread] processing WebGLCommand(WebGLContextId(1), DrawingBufferHeight(WebGLSender(..)), WebGLCommandBacktrace)
 0:05.51 pid:21232 [2020-03-20T16:09:10Z DEBUG canvas::webgl_thread] WebGLImpl::apply(DrawingBufferHeight(WebGLSender(..)))
 0:05.51 pid:21232 FAIL getError expected: NO_ERROR. Was INVALID_OPERATION : before draw
 0:05.51 pid:21232 [2020-03-20T16:09:10Z TRACE canvas::webgl_thread] processing WebGLCommand(WebGLContextId(1), DrawArraysInstanced { mode: 0, first: 0, count: 65536, primcount: 65536 }, WebGLCommandBacktrace)
 0:05.51 pid:21232 [2020-03-20T16:09:10Z DEBUG canvas::webgl_thread] WebGLImpl::apply(DrawArraysInstanced { mode: 0, first: 0, count: 65536, primcount: 65536 })
 0:17.33 TEST_END: TIMEOUT
```
so it looks like the call to `glDrawArraysInstanced` at https://github.com/servo/servo/blob/7ef4c65ac32071612ba2981d6fb5a6b2431c205d/components/canvas/webgl_thread.rs#L1713-L1718 isn't returning.

**Comment by asajeffrey** [comment]
Chat on #gfx: https://chat.mozilla.org/#/room/#gfx:mozilla.org/$lxy1osHiCS9avrIsQ2rAFE3YmtNKtdhS7ZN3mIg-gMw

TL;DR it's a stress test, we should probably be validating the size and fast-failing in this case.
